src/event.rs

```diff
@@ -0,0 +1,566 @@
+				Ok(Self::PaymentReceived { payment_hash, amount_msat })
+			}
+			3u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let user_channel_id: u64 = Readable::read(reader)?;
+				Ok(Self::ChannelReady { channel_id, user_channel_id })
+			}
+			4u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let user_channel_id: u64 = Readable::read(reader)?;
+				Ok(Self::ChannelClosed { channel_id, user_channel_id })
+			}
+			//5u8 => {
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful { payment_hash } => {
+				0u8.write(writer)?;
+				payment_hash.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentFailed { payment_hash } => {
+				1u8.write(writer)?;
+				payment_hash.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentReceived { payment_hash, amount_msat } => {
+				2u8.write(writer)?;
+				payment_hash.write(writer)?;
+				amount_msat.write(writer)?;
+				Ok(())
+			}
+			Self::ChannelReady { channel_id, user_channel_id } => {
+				3u8.write(writer)?;
+				channel_id.write(writer)?;
+				user_channel_id.write(writer)?;
+				Ok(())
+			}
+			Self::ChannelClosed { channel_id, user_channel_id } => {
+				4u8.write(writer)?;
+				channel_id.write(writer)?;
+				user_channel_id.write(writer)?;
+				Ok(())
+			} //Self::OnChainPaymentSent { .. } => {
+			  //TODO
+			  //}
+			  //Self::OnChainPaymentReceived { .. } => {
+			  //TODO
+			  //}
+		}
+	}
+}
+
+pub(crate) struct LdkLiteEventQueue<K: KVStorePersister> {
+	queue: Mutex<EventQueueSerWrapper>,
+	notifier: Condvar,
+	persister: Arc<K>,
+}
+
+impl<K: KVStorePersister> LdkLiteEventQueue<K> {
+	pub(crate) fn new(persister: Arc<K>) -> Self {
+		let queue: Mutex<EventQueueSerWrapper> = Mutex::new(EventQueueSerWrapper(VecDeque::new()));
+		let notifier = Condvar::new();
+		Self { queue, notifier, persister }
+	}
+	pub(crate) fn add_event(&self, event: LdkLiteEvent) -> Result<(), Error> {
+		{
+			let mut locked_queue = self.queue.lock().unwrap();
+			locked_queue.0.push_back(event);
+			self.persister.persist(EVENTS_PERSISTENCE_KEY, &*locked_queue)?;
+		}
+
+		self.notifier.notify_one();
+		Ok(())
+	}
+
+	pub(crate) fn next_event(&self) -> LdkLiteEvent {
+		let locked_queue = self
+			.notifier
+			.wait_while(self.queue.lock().unwrap(), |queue| queue.0.is_empty())
+			.unwrap();
+		locked_queue.0.front().unwrap().clone()
+	}
+
+	pub(crate) fn event_handled(&self) -> Result<(), Error> {
+		{
+			let mut locked_queue = self.queue.lock().unwrap();
+			locked_queue.0.pop_front();
+			self.persister.persist(EVENTS_PERSISTENCE_KEY, &*locked_queue)?;
+		}
+		self.notifier.notify_one();
+		Ok(())
+	}
+}
+
+impl<K: KVStorePersister> ReadableArgs<Arc<K>> for LdkLiteEventQueue<K> {
```

Comment:
We'll want a `Writeable` implementation, too.

Reply:
Why? `Readable`/`Writeable` is implemented for `EventQueueSerWrapper` and the `ReadableArgs` implementation of `EventQueue` merely a short form of reading the queue separately an passing it to something like `EventQueue::new`, which allows us to never expose `EventQueueSerWrapper`.